### PR TITLE
chore: adding e2e sections in cy configs that don't have them

### PR DESCRIPTION
**Pull request by elylucas**
This PR adds e2e config sections to projects that did not have them. This was causing the ci pipelines to break in Cy 10.1.0 as it now requires there be a e2e or component section when in run mode.

This PR also changes some of the examples to run with retries the the Cypress config instead of using the test-retries scripts. It was unclear on what the script was trying to achieve, and the tests were flaky, so I felt using the built in retries mechanism makes sense.

### CODE REVIEW

File: examples/blogs__assertion-counting/cypress.config.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
 const { defineConfig } = require('cypress')
 
-module.exports = defineConfig({})
+module.exports = defineConfig({
+  e2e: { },
```
Comment (jaffrepaul) [inline examples/blogs__assertion-counting/cypress.config.js:4]:
```suggestion
  e2e: {},
```

File: examples/stubbing-spying__intercept/cypress.config.js [hunk 2]
```diff
@@ -2,6 +2,10 @@ const { defineConfig } = require('cypress')
 
 module.exports = defineConfig({
   defaultCommandTimeout: 8000,
+  retries: {
+    runMode: 5,
+    openMode: 1,
```
Comment (jaffrepaul) [inline examples/stubbing-spying__intercept/cypress.config.js:7]: Just curious why the value for openMode as well here and not other projects?

**Review by jaffrepaul — CHANGES_REQUESTED** [review]
Added a couple tiny tweaks and a clarifying q 👍

File: examples/stubbing-spying__intercept/cypress.config.js (hunk 2 above)
Comment (elylucas) [inline examples/stubbing-spying__intercept/cypress.config.js:7]:
```suggestion
    openMode: 0,
```

**Review by jaffrepaul — COMMENTED** [review]
So just to clarify, our recommendation around this issue to include a testing type object in the config, even if it's empty? Curious why that creates issues without it.

File: examples/stubbing-spying__intercept/cypress.config.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ module.exports = defineConfig({
   defaultCommandTimeout: 8000,
   retries: {
     runMode: 5,
-    openMode: 1,
+    openMode: 0,
```
Comment (jaffrepaul) [inline examples/stubbing-spying__intercept/cypress.config.js:7]: It looks like the default is already "0" https://docs.cypress.io/guides/guides/test-retries#Global-Configuration
